Code/src/main/java/nl/utwente/group10/ui/components/FunctionBlock.java

```diff
@@ -31,32 +31,54 @@
 	 * @throws IOException
 	 */
 	public static FunctionBlock newInstance(int numberOfArguments){
-		//TODO Prettier resource loading
-		//TODO better factory
-		FunctionBlock functionBlock = null;
+		FunctionBlock functionBlock;
 		try{
 			functionBlock = (FunctionBlock) FXMLLoader.load(Main.class.getResource("/ui/FunctionBlock.fxml"), null, new TactileBuilderFactory());
 			functionBlock.initializeArguments(numberOfArguments);
 		}catch(IOException e){
 			e.printStackTrace();
+			functionBlock = null;
```

Comment:
I don't like this catch. All this will do is cause NullPointerExceptions down the line, so why catch it at all?